dds_web/api/project.py

```diff
@@ -169,33 +170,23 @@ def post(self):
                 )
             elif new_status == "Archived":
                 is_aborted = json_input.get("is_aborted", False)
+                is_queue_operation = True
                 new_status_row, delete_message = self.archive_project(
-                    project=project, current_time=curr_date, aborted=is_aborted
+                    project=project,
+                    current_time=curr_date,
+                    aborted=is_aborted,
                 )
             else:
                 raise DDSArgumentError(message="Invalid status")
 
-            try:
-                project.project_statuses.append(new_status_row)
-                project.busy = False  # TODO: Use set_busy instead?
-                db.session.commit()
-                flask.current_app.logger.info(
-                    f"Busy status set. Project: '{project.public_id}', Busy: False"
-                )
-            except (sqlalchemy.exc.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
-                flask.current_app.logger.exception(err)
-                db.session.rollback()
-                raise DatabaseError(
-                    message=str(err),
-                    alt_message=(
-                        "Status was not updated"
-                        + (
-                            ": Database malfunction."
-                            if isinstance(err, sqlalchemy.exc.OperationalError)
-                            else ": Server Error."
-                        )
-                    ),
-                ) from err
+            return_message = ""
+
+            if not is_queue_operation:
+                # If operations was not handled by a queue, commit the changes
+                self.update_project_status(project=project, new_status_row=new_status_row)
+                return_message += f"Project {project.public_id} updated to status {new_status}"
+            else:
+                return_message += f"The status of project {project.public_id} is being updated to {new_status}. The DDS is handling this in the background.  It may take some time to complete."
```

Comment:
I don't believe this line is tested, that we know it does print this, right?

Reply:
We have a bunch of tests for when errors happen, but I'd like to have a test that makes sure that specific status changes returns one thing and doesn't get added to a queue and that archiving returns another thing and does get added to the queue

Reply:
During live review: 
- The queuing etc is already indirectly tested, which is good, but I'd like us to add specific assertions to existing tests to verify that they are added to the queue / not added to the queue etc. Each project status transition should be tested like this.
- I'd like us to verify that this specific return_message is printed / returned, and also the same with the change in the `commands.py`